### DISCUSSION TITLE
Is it possible to watch for a class to be added to a div and run a javascript class when it happens?

### DISCUSSION THREAD

**Discussion by padders** [alpinejs/alpine#583] [4. General]
I'm trying to write some custom GSAP transitions and animations, and I was hoping I could get alpine to call the relevant function when a class is added to that div. Is there an x-on event to watch for a class and then call a JS function? Or a better method using alpine to do this? 

My scrolltrigger can call a function if needs be, or add a class to the div.

**Comment by padders** [comment]
For more info, I'm using this to trigger animations

https://greensock.com/docs/v3/Plugins/ScrollTrigger

I'd love to be able to pass the element reference in too if possible.

**Reply by SimoTod** [reply]
There's no javascript event to detect when a class is added. You can dispatch a custom event from your GSAP callback (on the document element) and then you can use x-on:custom-element-name.document="etc etc".
